Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -1415,78 +1505,57 @@ class DecoratorTransform(ScopeTrackingTransform, SkipDeclarations):
     with up to the three getter, setter and deleter DefNodes.
     The functional style isn't supported yet.
     """
-    _properties = None
+    _properties_node_sets = None
 
-    _map_property_attribute = {
+    _property_attributes = {
         'getter': EncodedString('__get__'),
         'setter': EncodedString('__set__'),
         'deleter': EncodedString('__del__'),
-    }.get
+    }
+
+    _map_property_attribute = _property_attributes.get
 
     def visit_CClassDefNode(self, node):
-        if self._properties is None:
-            self._properties = []
-        self._properties.append({})
+        if self._properties_node_sets is None:
+            self._properties_node_sets = []
+        id_props_transform = _IdentifyPropertiesTransform(self.context,
```

Comment:
Does this require a new transform? Can it not be done along the way?

Reply:
The reason for doing it separately was for things like:

```
cdef class C:
    @property
    def x(self):
       return self._x
    @x.setter
    @some_other_decorator
    def x(self, value):
        self._x = value
```

Because of `some_other_decorator` `x` can't be a `cdef property` (but can be handled by the python property function). It isn't possible to know this at the point that the getter is evaluated though.

Therefore I think it does need two passes - one to collect the properties and one clear any functions that have been changed to properties.

They should both be pretty shallow transforms though, since `cdef class`es can't be nested so there's no need to actually go into the functions.